lib/diplomat/value.ex

```diff
@@ -14,85 +19,102 @@ defmodule Diplomat.Value do
 
   def new(val), do: %__MODULE__{value: val}
 
-  def from_proto(%PbVal{boolean_value: val}) when is_boolean(val),
-    do: new(val)
-  def from_proto(%PbVal{integer_value: val}) when is_integer(val),
-    do: new(val)
-  def from_proto(%PbVal{double_value: val}) when is_float(val),
-    do: new(val)
-  def from_proto(%PbVal{string_value: val}) when is_binary(val),
-    do: new(val)
-  def from_proto(%PbVal{blob_value: val}) when is_bitstring(val),
-    do: new(val)
-  def from_proto(%PbVal{timestamp_microseconds_value: val}) when is_integer(val),
-    do: new(val) # need to convert this to a timestamp of some sort
-  def from_proto(%PbVal{key_value: val}) when not is_nil(val) do
+  def from_proto(%PbVal{value_type: {:boolean_value, val}}) when is_boolean(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:integer_value, val}}) when is_integer(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:double_value, val}}) when is_float(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:string_value, val}}) when is_binary(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:blob_value, val}}) when is_bitstring(val) do
+    new(val)
+  end
```

Comment:
For all these `from_proto/1` methods, I'd prefer we keep them written in the same style they were before, something similar to:

```elixir
def from_proto(val),
  do: new(val)
```

Instead of:
```elixir
def from_proto(val) do
  new(val)
end
```